src/events/client/interactionCreate.ts

```diff
@@ -0,0 +1,80 @@
+
+export default async function interactionCreateHandler(interaction: Eris.PingInteraction | Eris.CommandInteraction | Eris.ComponentInteraction | Eris.UnknownInteraction) {
+    if (interaction instanceof Eris.ComponentInteraction) {
+        const gameSession = state.gameSessions[interaction.guildID];
+        if (!gameSession || !gameSession.gameRound) {
+            return;
+        }
+
+        const messageContext = new MessageContext(
+            interaction.channelID,
+            new KmqMember(interaction.member.username, getUserTag(interaction.member), interaction.member.avatarURL, interaction.member.id),
+            interaction.guildID,
+        );
+
+        if (gameSession.gameRound.incorrectMCGuessers.has(interaction.member.id)) {
+            await interaction.createMessage({
+                embeds: [{
+                    color: EMBED_ERROR_COLOR,
+                    author: {
+                        name: messageContext.author.username,
+                        icon_url: messageContext.author.avatarUrl,
+                    },
+                    title: bold("Invalid guess"),
+                    description: "You've already been eliminated for this round.",
+                    thumbnail: { url: KmqImages.DEAD },
+                }],
+                flags: 64,
+            });
+            return;
+        }
+
+        if (!gameSession.isValidInteractionGuess(interaction.data.custom_id)) {
+            await interaction.createMessage({
+                embeds: [{
+                    color: EMBED_ERROR_COLOR,
+                    author: {
+                        name: messageContext.author.username,
+                        icon_url: messageContext.author.avatarUrl,
+                    },
+                    title: bold("Invalid guess"),
+                    description: "You are attempting to pick an option from an already completed round.",
+                    thumbnail: { url: KmqImages.DEAD },
+                }],
+                flags: 64,
+            });
+            return;
+        }
+
+        if (!gameSession.isCorrectInteractionAnswer(interaction.data.custom_id)) {
+            gameSession.gameRound.incorrectMCGuessers.add(interaction.member.id);
+            await interaction.createMessage({
+                embeds: [{
+                    color: EMBED_ERROR_COLOR,
+                    author: {
+                        name: messageContext.author.username,
+                        icon_url: messageContext.author.avatarUrl,
+                    },
+                    title: bold("Incorrect guess"),
+                    description: "You've been eliminated for this round.",
+                    thumbnail: { url: KmqImages.DEAD },
+                }],
+                flags: 64,
+            });
+            gameSession.guessSong(messageContext, "");
+            return;
+        }
+
+        const guildPreference = await getGuildPreference(messageContext.guildID);
```

Comment:
Missing ack?

Reply:
The ack will be the editParent inside guessSong